WordPress/gutenberg
# Fix screen flash when deleting templates in templates list

## What?
Fixes https://github.com/WordPress/gutenberg/issues/37926.

Without this PR, there is a screen flash when deleting templates from the templates list – an unnecessary disruption of UX.

## How?
I made the templates list always visible when `templates` are available, even if (temporarily) not up-to-date.

Technically, I removed the `isLoading` condition for not displaying the templates list. Hiding the table is only useful on the initial load when `templates` is already `null`. 

I don't think it makes sense to display a loader while re-fetching the data. It happens fast enough that hiding the table to display a loader would also look like a screen flash.

## Testing Instructions

1. Create a new template from the Site Editor.
2. Head to the template list and delete it.
3. See no flash on the screen when the request is in progress.

cc @annezazu @youknowriad @adamziel
